### ISSUE TITLE
Internal error and authentication failed durring sync on Windows 10

### ISSUE THREAD

**Issue by mattish91** [PascalLG/Nubo#3]
Hi, after about 21 uploaded files in a folder i get "error: authentication failed" and "error: unexpected server error (internal server error ([path to my nubo install]\php\sqlite.php 93))"

I would love to help resolving this, im using SSL
on a Windows Server 2016 using Apache 2.4 + php 7.1.14-Win32-VC14
incase thats to any help.

Thanks!

Feedback: If possible, add a script to run in the end of the install to configure, also set path to sync. Great work!

**Comment by PascalLG** [comment]
Hi, do you get these errors in the middle of a synchronisation, while running a `nubo sync` command? If so, does the file being synchronised at the time of the error (or the one just before) have anything special? I mean unusual attributes set, or a very long filename, or a filename with non-ANSI characters, for example.

I never experienced this error myself, by I guess it could happen in case of a network failure in the middle of a synchronisation. You can authenticate again using `nubo auth` and try to synchronise again. Tell me if the error persists.

Also, did you run `nubo init` with the `--test` flag at installation? This could be useful to determine whether the server behaves properly or not.

Thank you for your report and feedback!

**Comment by mattish91** [comment]
yes, the files are named as ".[random numbers and letters and symbols]" i don't know where those files came from tho, i have checked hidden files but can't see them. And the files are not syncronizing after the first error message appers. I did nubo init --test and that works just fine.

**Comment by PascalLG** [comment]
I improved the error reporting on the server side to help debugging this kind of problem. You can download an installer for this modified version [here](https://raw.githubusercontent.com/PascalLG/Nubo/master/Prod/nubo.php). (Just save this file on your computer as nubo.php, upload it on your server, and install as usual.) Can you post here the error message you get when synchronising with this new server version?

**Comment by mattish91** [comment]
oh crap, it started to remove some of my files now :O I had to stop it instantly to not wipe everything that i wan't to sync :O
this is my output:
Error: Project\avfuktningsgruppen.se\bilder.zip (file is too big)
delete local: Antagande\Dokumentation._0001.txt
delete local: Dokumentation\Dokumentation_0002.txt
delete local: Dokumentation\Prisförslag.txt
delete local: Dokumentation\Referens bilder\hdd_fail_0001.png
delete local: Faktura\Faktura_0001.pdf
delete local: Grav Backup\ mspot-nu-20170909135343.zip
send: Grav Backup\ mspot-nu-20171006221326.zip
19 % |#########-----------------------------------------|    error: Unexpected error (InternalException Network.Socet.recvBuf: failed (no error))
send: Grav Backup\ mspot-nu-20171007211420.zip
100 % |##################################################| error: internal server error (http 500)
send: Grav Backup\ mspot-nu-2017112214214.zip

and the worse is it actually deleted the files :O (i do have a backup) this was just test environment luckily phew ^^ i hope this helps in any way.

**Comment by mattish91** [comment]
Did this solve anything? i just installed it again and synced some files between my laptop and my server, seams to work just fine. is there any way to make the sync happen instantly if there is new files? without commands ^^

**Comment by PascalLG** [comment]
I improved a few things since December, but I did not merge these changes to master branch yet. I have to perform more testings first, specially on Windows. (And I added a safety feature to avoid mistakenly removing files when the server is updated!) 

Actually the command-triggered sync is by design :) I often work on apps or scripts that generate dozens of temporary files and I find it very irritating when your cloud spends most of its time synchronising useless files that won't even exist anymore minutes later. However, I'm sure one can write a small Win32 application that registers for file system events and launches nubo every time it detects a file is added/modified/deleted in a specific folder. (Same for Linux and macOS, BTW.)
